### PR TITLE
Adds Art Collecting [Ivniinvi PR recoup]

### PR DESCRIPTION
**Pull request by PestoVerde322**
## About The Pull Request

This PR adds art collecting. There is now a verb in the OOC tab that lets you browse the art gallery. You can then buy any art piece from its owner for a price of 500 beecoins + 500 for every time it's changed hands. 1/4 of the sale price goes to the original artist, 1/4 goes to the person you bought it from, and 1/2 goes to the void. Additionally, if art is bought during the pregame, it sends a message in OOC describing the purchase. Each painting's initial owner is the artist.

Purchasing art is strictly nonprofitable, you will never make positive beecoins off of purchasing art. At best, you will make back slightly more than a quarter of the purchase price.

(this is a PR done by Ivniinvi https://github.com/BeeStation/BeeStation-Hornet/pull/6895).

## Why It's Good For The Game

This can act as a proto-stock market that can be used to reclaim *some* beecoins, and incentivize the creation of art content.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

immagine
_(Text is from the original PR which is incorrect, and has since been corrected)_

</details>

## Changelog
:cl:Ivniinvi (recouped and updated by PestoVerde322)
add: Art collecting! Use the Browse Art Gallery verb in the OOC tab to purchase art. The original artist gets 1/4 of the purchase price, and the current owner gets 1/4.
/:cl:

### CODE REVIEW

**Comment by pixelman77** [comment]
NFTs invading bee

**Comment by DerpiestBruh** [comment]
Real.

File: code/controllers/subsystem/persistence.dm [hunk 1]
```diff
@@ -16,6 +16,7 @@ SUBSYSTEM_DEF(persistence)
 	var/list/obj/structure/sign/painting/painting_frames = list()
 	var/list/obj/item/storage/photo_album/photo_albums
 	var/list/paintings = list()
```
Comment (PowerfulBacon) [inline code/controllers/subsystem/persistence.dm:18]: Paintings needs to be protected var if it contains information about the price and owner

File: code/controllers/subsystem/persistence.dm [hunk 2]
```diff
@@ -294,6 +295,42 @@ SUBSYSTEM_DEF(persistence)
 	for(var/obj/structure/sign/painting/P in painting_frames)
 		P.load_persistent()
 
+	json_file = file("data/pending_art_metacoin.json")
+	if(fexists(json_file))
+		pending_art_metacoin = json_decode(file2text(json_file))
+
+	if(paintings && length(paintings["library"]) && !paintings["library"][1]["price"]) //this should only ever happen once - adding owner and price values to legacy paintings
```
Comment (PowerfulBacon) [inline code/controllers/subsystem/persistence.dm]: I feel like this should be able to detect legacy paintings always, in the case that testmerge is applied / removed or this is reverted. Perhaps check the last painting ratehr than the first one?

File: code/datums/art_browser.dm [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+/datum/art_browser/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+
+	var/list/chosen_portrait = SSpersistence.paintings["library"][params["selected"]]
+	var/author = chosen_portrait["ckey"]
+	var/title = chosen_portrait["title"]
+	var/price = chosen_portrait["price"]
+	var/owner = chosen_portrait["owner"]
```
Comment (PowerfulBacon) [inline code/datums/art_browser.dm:43]: Should have a check to ensure chosen_portrait isn't null

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by itsmeow** [comment]
@PestoVerde322 Revive?

**Comment by PestoVerde322** [comment]
> @PestoVerde322 Revive?

not 100% sure, since i got busy with other stuff, might start to revive it and get to the reviews bacon pointed

File: code/controllers/subsystem/persistence.dm (hunk 1 above)
Comment (PestoVerde322) [reply code/controllers/subsystem/persistence.dm:18]: Tried to slap the good ol'reliable "protection = CONFIG_ENTRY_LOCKED" but then i remember it's tied to configurations rather subsystems, so i'm kinda stumped on that one...

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Check vv_edit for ways to add protection from admins editting via varedit. Nothing being saved to the server should be able to be vareditted

**Review by PowerfulBacon — APPROVED** [review]
Apparently there is no way to protect lists from admin edits. Until we rework this, we may have to just remove any admins abusing their power.

File: tgui/packages/tgui/interfaces/ArtGallery.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
+import { resolveAsset } from '../assets';
+import { useBackend, useLocalState } from '../backend';
+import { Button, NoticeBox, Section, Stack, Tabs } from '../components';
+import { Window } from '../layouts';
+
+export const ArtGallery = (props, context) => {
```
Comment (itsmeow) [inline tgui/packages/tgui/interfaces/ArtGallery.js:6]: pretty sure this UI is mostly copypasted from the ModPC art gallery. Use components properly and integrate the art gallery app under one component

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Pull request closed without merging**